script.js

```diff
@@ -1,43 +1,57 @@
-const peopleInSpace = document.querySelector("[data-js='people-in-space']");
-const listOfPeople = document.querySelector("[data-js='list-of-people']");
-const allButton = document.querySelector("[data-js='all-button']");
-const issButton = document.querySelector("[data-js='iss-button']");
-const tiangongButton = document.querySelector("[data-js='tiangong-button']");
+const peopleInSpaceElement = document.querySelector(
+  "[data-js='people-in-space']"
+);
+const listOfPeopleElement = document.querySelector(
+  "[data-js='list-of-people']"
+);
+const allButtonElement = document.querySelector("[data-js='all-button']");
+const issButtonElement = document.querySelector("[data-js='iss-button']");
+const tiangongButtonElement = document.querySelector(
+  "[data-js='tiangong-button']"
+);
 
-// function to creating list on Dom
+// Create list on the DOM
 function updatePeopleList(peopleArray) {
+  listOfPeopleElement.innerHTML = "";
+
   peopleArray.forEach((person) => {
     const listElement = document.createElement("li");
     listElement.textContent = person.name;
-    listOfPeople.appendChild(listElement);
-    peopleInSpace.textContent = peopleArray.length;
+    listOfPeopleElement.appendChild(listElement);
   });
+
+  peopleInSpaceElement.textContent = peopleArray.length;
 }
 
-async function getPeopleInSpace() {
+// Fetch data
+async function fetchPeopleInSpace() {
   const response = await fetch("http://api.open-notify.org/astros.json");
-  const data = await response.json();
-
-  updatePeopleList(data.people);
+  const { people } = await response.json();
+  return people;
+}
 
-  issButton.addEventListener("click", () => {
-    const issPeople = data.people.filter((person) => person.craft === "ISS");
-    listOfPeople.innerHTML = "";
-    updatePeopleList(issPeople);
-  });
+// Event handler function
+function handleButtonClick(data, craftType) {
+  const filteredPeople = craftType
```

Comment:
If `data` is always an array, then i don't think we will ever reach the `else` part of the ternary operator. (i.e. even an empty array is truthy).

Reply:
I am not sure I understand your statement. 
Not data acts as a condition here but craftType. We always filter the available data based on the craftType. If set we filter the data accordingly, otherwise we return the entire data set (as we are interested in all astronauts on all space crafts).

Reply:
You are perfectly right! My mistake.